### ISSUE TITLE
CLI generating incomplete Unauth (Guest) IAM Policy for API Gateway

### ISSUE THREAD

**Issue by warrenmcquinn** [aws-amplify/amplify-cli#1905]
**Describe the bug**
Amplify CLI creates (or updates) two policies when adding unauthenticated (Guest) access to REST API routes.

The **auth** policy is titled `PolicyAPIGW` + REST API resource name + `auth`.
The **unauth** policy is titled `PolicyAPIGW` + REST API resource name + `unauth`.

For example, I'll add an open GET method endpoint called `/test`:
```
$ amplify api update                                                                                                                                                                    
? Please select from one of the below mentioned services REST
? Please select the REST API you would want to update rest
? What would you like to do Add another path
? Provide a path (e.g., /items) /test
? Choose a Lambda source Create a new Lambda function
? Provide a friendly name for your resource to be used as a label for this category in the project: XXX
? Provide the AWS Lambda function name: XXX
? Choose the function template that you want to use: Serverless express function (Integration with Amazon API Gateway)
? Do you want to access other resources created in this project from your Lambda function? No
? Do you want to edit the local lambda function now? No
Succesfully added the Lambda function locally
? Restrict API access Yes
? Who should have access? Authenticated and Guest users
? What kind of access do you want for Authenticated users? create, read, update, delete
? What kind of access do you want for Guest users? create, read, update, delete
? Do you want to add another path? No
Successfully updated resource
```

The **auth** policy allows `execute-api:Invoke` on at least two APIGW resources:
GET `/test`
GET `/test/*`

However, the **unauth** policy only allows `execute-api:Invoke` on:
GET `/test/*`

Expected behavior: **unauth** policy should grant `execute-api:Invoke` on the same resources as the **auth** policy. 

CLI Version 1.8.5

**Comment by warrenmcquinn** [comment]
Had some free time today so I took a crack at a fix in #1906. Tested locally using amplify-dev. Happy with the results! First OSS contribution so please be gentle, but I'm open to feedback 👍

**Comment by warrenmcquinn** [comment]
Fix merged in [bcd0d02](https://github.com/aws-amplify/amplify-cli/commit/bcd0d02a229d3dab2e5babc40b68ac9090aa5f15)
